src/Morphir/TypeScript/Backend.elm

```diff
@@ -81,93 +83,155 @@ mapModuleDefinition opt distribution currentPackagePath currentModulePath access
             , typeDefs = typeDefs
             }
     in
-    [moduleUnit]
+    [ moduleUnit ]
+
+
+{-| Map a Morphir Constructor (A tuple of Name and Constructor Args) to a Typescript AST Interface
+-}
+mapConstructor : TS.Privacy -> List TS.TypeExp -> ( Name, List ( Name, Type.Type ta ) ) -> TS.TypeDef
+mapConstructor privacy variables ( ctorName, ctorArgs ) =
+    let
+        nameInTitleCase =
+            ctorName |> Name.toTitleCase
+
+        kindField =
+            ( "kind", TS.LiteralString nameInTitleCase )
+
+        otherFields =
+            ctorArgs
+                |> List.map
+                    (\( argName, argType ) ->
+                        ( argName |> Name.toCamelCase, mapTypeExp argType )
+                    )
+    in
+    TS.Interface
+        { name = nameInTitleCase
+        , privacy = privacy
+        , variables = variables
+        , fields = (kindField :: otherFields)
+        }
 
 
 {-| Map a Morphir type definition into a list of TypeScript type definitions. The reason for returning a list is that
 some Morphir type definitions can only be represented by a combination of multiple type definitions in TypeScript.
 -}
 mapTypeDefinition : Name -> AccessControlled (Documented (Type.Definition ta)) -> List TS.TypeDef
 mapTypeDefinition name typeDef =
+    let
+        doc =
+            typeDef.value.doc
+
+        privacy =
+            case typeDef.access of
+                Public ->
+                    TS.Public
+
+                Private ->
+                    TS.Private
+    in
     case typeDef.value.value of
-        Type.TypeAliasDefinition typeArgs typeExp ->
+        Type.TypeAliasDefinition variables typeExp ->
             [ TS.TypeAlias
-                (name |> Name.toTitleCase)
-                (typeExp |> mapTypeExp)
+                { name = (name |> Name.toTitleCase)
```

Comment:
The parentheses around `name |> Name.toTitleCase` are unnecessary, and should have been removed by autoformatting. This doesn't match up with the current head of dwinship/Typescript-Backend.

There's a few other autoformatting changes between this branch and the unsquashed branch.

(I think I may have confused things during a rebase in the last couple of days. I think I added the autoformatting changes, but edited the changes into one of the existing commits, rather than giving them their own commit. That might have contributed to this.)

Reply:
Huh, that's weird. I was wrong about dwinship/Typescript-Backend having all the autoformat fixes. It had some, but not all of them.

I've rebased and updated both of them to have all the autoformatting now, and the Typescript Backend files are identical on both branches now.

Reply:
Confirmed that `elm-format` doesn't make any changes on commit ccfc7b8bb9b890132c5920c43554db5b5a801953. I probably introduced this problem when I squashed and rebased your work, thanks for fixing.